go/cli/mcap/cmd/merge.go

```diff
@@ -157,6 +169,13 @@ func (m *mcapMerger) mergeInputs(w io.Writer, inputs []namedReader) error {
 	profiles := make([]string, len(inputs))
 	pq := utils.NewPriorityQueue(nil)
 
+	// Reset struct members
+	m.schemaIdByHash = make(map[string]uint16)
+	m.schemaIDs = make(map[schemaID]uint16)
+	m.channelIDs = make(map[channelID]uint16)
+	m.nextChannelID = 1
+	m.nextSchemaID = 1
```

Comment:
Here I have to reset the `schemaIdByHash` as we might have a new writer where the schemas haven't been written to yet. 
In general, I don't fully understand the purpose of the mcapMerger struct. Wouldn't a single function be sufficient? I don't think users are supposed to call `mcapMerger.addSchema` themselves? But maybe I don't fully understand Go's concept of private members yet.

Reply:
I think this could definitely be written as a single function. There are no users of mcapMerger since it's private. I don't remember the details about mcapMerger but I'm pretty sure it only exists to assist with whatever remapping of channel/schema IDs are required, and you could write that either way.

Reply:
are you doing this resetting because you anticipate mergeInputs will be called multiple times?

Reply:
^^ this seems reasonable to me (though I don't think it actually will be called that way), as would writing a single function -- maybe that's why you were asking.

Patch looks good to me generally, thanks!

Reply:
> are you doing this resetting because you anticipate mergeInputs will be called multiple times?

It's [called multiple times](https://github.com/foxglove/mcap/blob/e422d7f9917be88644c401f89368c117bf105eda/go/cli/mcap/cmd/merge_test.go#L145-L153) in the tests. The resetting was required to make it succeed.